lib/Map/RegionProviderTs.ts

```diff
@@ -0,0 +1,42 @@
+import RegionProvider from "./RegionProvider";
+import isDefined from "../Core/isDefined";
+
+/**
+ * Apply an array of regular expression replacements to a string. Also caches the applied replacements in regionProvider._appliedReplacements.
+ * @private
+ * @param {RegionProvider} regionProvider The RegionProvider instance.
+ * @param {String} s The string.
+ * @param {String} replacementsProp Name of a property containing [ [ regex, replacement], ... ], where replacement is a string which can contain '$1' etc.
+ */
+export function applyReplacements(
+  regionProvider: RegionProvider,
+  s: string | number,
+  replacementsProp: "dataReplacements" = "dataReplacements"
```

Comment:
The type here doesn't make sense. If the only possible value of replacementsProp is `"dataReplacements"`, then why make it a parameter at all? I'm guessing it should be a string.

Reply:
Type should have been ` "dataReplacements"   | "serverReplacements"  | "disambigDataReplacements"`